base/multidimensional.jl

```diff
@@ -1638,8 +1638,9 @@ julia> unique(A, dims=3)
 ```
 """
 unique(A::AbstractArray; dims::Union{Colon,Integer} = :) = _unique_dims(A, dims)
+unique(A::AbstractVector; dims::Union{Colon,Integer} = 1) = _unique_dims(A, dims)
 
-_unique_dims(A::AbstractArray, dims::Colon) = invoke(unique, Tuple{Any}, A)
+_unique_dims(A::AbstractArray, dims::Colon) = _unique_dims(vec(A), 1)
```

Comment:
This routes `unique(::Vector)` to the `_unique_dims` code path which is much slower than the `unique(itr)` codepath

```julia
julia> using Chairmarks

julia> x = rand(1:100, 1000);

julia> @b x unique
3.861 μs (15 allocs: 5.250 KiB)

julia> @eval Base begin
       unique(A::AbstractVector; dims::Union{Colon,Integer} = 1) = _unique_dims(A, dims)
       _unique_dims(A::AbstractArray, dims::Colon) = _unique_dims(vec(A), 1)
       end
_unique_dims (generic function with 2 methods)

julia> @b x unique
9.583 μs (29 allocs: 24.938 KiB)
```